optimizely/optimizely.py

```diff
@@ -63,20 +63,21 @@ def __init__(self,
 
     try:
       self.config = project_config.ProjectConfig(datafile, self.logger, self.error_handler)
-    except:
+    except exceptions.UnsupportedDatafileVersionException as error:
       self.is_valid = False
-      self.config = None
       # We actually want to log this error to stderr, so make sure the logger
       #   has a handler capable of doing that.
       self.logger = _logging.reset_logger(self.logger_name)
-      self.logger.error(enums.Errors.INVALID_INPUT_ERROR.format('datafile'))
+      self.logger.exception(str(error))
       return
 
-    if not self.config.was_parsing_successful():
+    except:
       self.is_valid = False
+      self.config = None
       # We actually want to log this error to stderr, so make sure the logger
       #   has a handler capable of doing that.
-      self.logger.error(enums.Errors.UNSUPPORTED_DATAFILE_VERSION)
+      self.logger = _logging.reset_logger(self.logger_name)
+      self.logger.error(enums.Errors.INVALID_INPUT_ERROR.format('datafile'))
```

Comment:
Can we check the type of the exception, and if it's `exceptions.UnsupportedDatafileVersionException`, then we log differently? I'd like that better than having two `except` clauses here. 

Also, we should pass the exception to the `error_handler` instance